h2o-r/h2o-package/R/frame.R

```diff
@@ -3227,7 +3227,10 @@ as.h2o <- function(x, destination_frame="", ...) {
 #' @method as.h2o default
 #' @export
 as.h2o.default <- function(x, destination_frame="", ...) {
-  if( destination_frame=="" ) destination_frame <- deparse(substitute(x)) # guessing is done in as.h2o.data.frame
+  if( destination_frame=="" ) {
+    subx <- destination_frame.guess(deparse(substitute(x)))
+    destination_frame <- .key.make(if(nzchar(subx)) subx else paste0(class(x), "_"))
```

Comment:
`paste0(class(x), "_"))` might ended up being of length=2, use `class(x)[1]` or `paste(collapse=`

Reply:
good point, thx, didn't actually know `class(x)` could return a list, will use `collapse`

Reply:
not a list but a character vector :)